src/main/java/com/avispl/symphony/dal/avdevices/camera/panasonic/awue150/CameraPanasonicAWUE150Communicator.java

```diff
@@ -0,0 +1,1779 @@
+	 * @param stats store all statistics
+	 * @param advancedControllableProperties store all controllable properties
+	 */
+	private void populateGainControl(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties) {
+		Integer currentGainControlValue;
+		Integer minGainValue = Integer.parseInt(Gain.G_0.getUiName());
+		Integer maxGainValue = Integer.parseInt(Gain.G_42.getUiName());
+		try {
+			currentGainControlValue = Integer.parseInt(cachedLiveCameraInfo.getGain().getUiName());
+		} catch (Exception e) {
+			currentGainControlValue = minGainValue;
+		}
+
+		String propertyName = DevicesMetricGroup.IMAGE_ADJUST.getName() + DeviceConstant.HASH + ImageAdjustControlMetric.GAIN.getName();
+
+		addAdvanceControlProperties(advancedControllableProperties,
+				createSlider(stats, propertyName, minGainValue.toString(), maxGainValue.toString(), minGainValue.floatValue(), maxGainValue.floatValue(), currentGainControlValue.floatValue()));
+		stats.put(propertyName.concat(ImageAdjustControlMetric.CURRENT_VALUE.getName()), currentGainControlValue.toString());
+	}
+
+	/**
+	 * This method is used to handle image adjust control
+	 *
+	 * @param stats store all statistics
+	 * @param advancedControllableProperties store all controllable properties
+	 * @param controllableProperty controllable property
+	 * @param value value of controllable property
+	 */
+	private void imageAdjustControl(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties, String controllableProperty, String value) throws FailedLoginException {
+		String groupName = DevicesMetricGroup.IMAGE_ADJUST.getName() + DeviceConstant.HASH;
+		ImageAdjustControlMetric imageAdjustControlMetric = ImageAdjustControlMetric.getByName(controllableProperty);
+		isEmergencyDelivery = true;
+		switch (imageAdjustControlMetric) {
+			case IRIS:
+				String currentValueInHex = convertFromUIValueToAPIValue(Float.parseFloat(value), DeviceConstant.MAX_IRIS_API_VALUE, DeviceConstant.MIN_IRIS_API_VALUE_WITH_2160P_FORMAT,
+						DeviceConstant.MAX_IRIS_UI_VALUE,
+						DeviceConstant.MIN_IRIS_UI_VALUE_WITH_2160P_FORMAT);
+
+				String command = Command.HASH.concat(Command.IRIS).concat(currentValueInHex.toUpperCase());
+				String responseValueInHex = performCameraControl(command, controllableProperty, value, DeviceURL.CAMERA_PTZ_CONTROL, true);
+				cachedLiveCameraInfo.setIrisPosition(responseValueInHex.substring(Command.IRIS_POSITION_RESPONSE.length()));
+
+				// camera need 3s to apply new value
+				try {
+					Thread.sleep(3000);
+				} catch (InterruptedException e) {
+					e.printStackTrace();
+				}
+				retrieveSimultaneous();
+				populateIrisControl(stats, advancedControllableProperties);
+				break;
+			case GAIN:
+				Gain gainControlValue;
+				try {
+					gainControlValue = Gain.getByUIName(String.valueOf((int) Float.parseFloat(value)));
+
+				} catch (Exception e) {
+					gainControlValue = cachedLiveCameraInfo.getGain();
```

Comment:
Would it make sense to have a debug/warn entry logged here? Just to track down that exception was caught here

Reply:
I fixed it